lib/gurney/source/npm.rb

```diff
@@ -0,0 +1,54 @@
+require 'json'
+
+module Gurney
+  module Source
+    class Npm < Base
+
+      SUPPORTED_LOCKFILE_VERSIONS = [2, 3].freeze
+     
+      def initialize(package_lock_json:)
+        @package_lock_json = package_lock_json
+      end
+
+      def present?
+        !@package_lock_json&.empty?
+      end
+
+      def dependencies
+        if present?
+          parsed_lock = JSON.parse(@package_lock_json)
+
+          if SUPPORTED_LOCKFILE_VERSIONS.include?(parsed_lock['lockfileVersion'])
+            extract_dependencies(parsed_lock)
+          else
+            raise Gurney::Error.new("package-lock.json: lockfileVersion #{parsed_lock['lockfileVersion']} is not supported")
```

Comment:
Shall we just print a warning and not crash? I'd rather have some dependencies reported than none at all.

(Would be something like `puts "package-lock.json: lockfileVersion #{parsed_lock['lockfileVersion']} is not supported; will not report npm dependencies.".yellow`` with a `require 'colorize'` at the top.)